absl/types/CMakeLists.txt

```diff
@@ -130,7 +130,7 @@ absl_test(
 
 # test any_exception_safety_test
 set(ANY_EXCEPTION_SAFETY_TEST_SRC "any_exception_safety_test.cc")
-set(ANY_EXCEPTION_SAFETY_TEST_PUBLIC_LIBRARIES absl::any absl::base)
+set(ANY_EXCEPTION_SAFETY_TEST_PUBLIC_LIBRARIES absl::any absl::base absl::bad_any_cast absl::strings)
```

Comment:
I think the bad_any_cast dep needs to be on absl::any.  This test also doesn't depend on absl::strings as far as I can tell

Reply:
You are right "bad_any_cast" has "any" and "base" as a dep. But without "absl::strings" it does not compile:

CMakeFiles/any_exception_safety_test_bin.dir/any_exception_safety_test.cc.o: In function `absl::substitute_internal::Arg::Arg(int)':
any_exception_safety_test.cc:(.text._ZN4absl19substitute_internal3ArgC2Ei[_ZN4absl19substitute_internal3ArgC5Ei]+0x25): undefined reference to `absl::numbers_internal::FastIntToBuffer(int, char*)'
CMakeFiles/any_exception_safety_test_bin.dir/any_exception_safety_test.cc.o: In function `absl::SubstituteAndAppend(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, absl::string_view, absl::substitute_internal::Arg const&, absl::substitute_internal::Arg const&)':
any_exception_safety_test.cc:(.text._ZN4absl19SubstituteAndAppendEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEENS_11string_viewERKNS_19substitute_internal3ArgESB_[_ZN4absl19SubstituteAndAppendEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEENS_11string_viewERKNS_19substitute_internal3ArgESB_]+0x6e): undefined reference to `absl::substitute_internal::SubstituteAndAppendArray(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, absl::string_view, absl::string_view const*, unsigned long)'
CMakeFiles/any_exception_safety_test_bin.dir/any_exception_safety_test.cc.o: In function `testing::Message& testing::Message::operator<< <absl::string_view>(absl::string_view const&)':
any_exception_safety_test.cc:(.text._ZN7testing7MessagelsIN4absl11string_viewEEERS0_RKT_[_ZN7testing7MessagelsIN4absl11string_viewEEERS0_RKT_]+0x35): undefined reference to `absl::operator<<(std::ostream&, absl::string_view)'

Reply:
That looks to be coming from absl/base/internal/exception_Safety_testing.  I have an internal change adding that dep there.